crates/cubecl-linalg/src/tensor/contiguous.rs

```diff
@@ -114,6 +145,45 @@ fn into_contiguous_kernel<N: CubePrimitive>(
     }
 }
 
+#[cube(launch)]
+fn into_contiguous_kernel_pack<N: CubePrimitive>(
+    input: &Tensor<Line<N>>,
+    output: &mut Tensor<Line<N>>,
+    out_layout: StridedLayout,
+    shape: Sequence<FastDivmod>,
+    stride: Sequence<u32>,
+    #[comptime] elems_per_thread: u32,
+) {
+    let line_size = output.line_size();
+    let lines_per_thread = comptime![elems_per_thread / line_size];
+
+    let offset_output = ABSOLUTE_POS * lines_per_thread;
+    let offset_input = offset_output * line_size;
+
+    let mut registers = Array::<Line<N>>::vectorized(lines_per_thread, line_size);
+
+    #[unroll]
+    for i in 0..lines_per_thread {
+        let offset = i * line_size;
+        let mut reg = Line::<N>::empty(line_size);
+        #[unroll]
+        for k in 0..line_size {
+            let offset_input = offset_input + offset + k;
+            let offset_input =
+                index_offset_contiguous_fastdivmod::<N>(input, offset_input, &shape, &stride);
+            reg[k] = input[offset_input][0];
```

Comment:
I guess it's better to not vectorize the input?

Reply:
If the input can be vectorized it's still better to just do 1 to 1 mapping but it's rare. the `pack` kernel is specifically for the case where the input can't be vectorized, to ensure writes are more efficient.